packages/main/src/plugin/image-registry.spec.ts

```diff
@@ -496,3 +496,92 @@ describe('expect checkCredentials', async () => {
     expect(registries2.length).toBe(1);
   });
 });
+
+test('getBestManifest returns the expected manifest', () => {
+  const manifests = {
+    'linux-amd64': {
+      platform: {
+        architecture: 'amd64',
+        os: 'linux',
+      },
+      name: 'linux-amd64',
+    },
+    'linux-arm64': {
+      platform: {
+        architecture: 'arm64',
+        os: 'linux',
+      },
+      name: 'linux-arm64',
+    },
+    'windows-amd64': {
+      platform: {
+        architecture: 'amd64',
+        os: 'windows',
+      },
+      name: 'windows-amd64',
+    },
+    'windows-arm64': {
+      platform: {
+        architecture: 'arm64',
+        os: 'windows',
+      },
+      name: 'windows-arm64',
+    },
+    'darwin-amd64': {
+      platform: {
+        architecture: 'amd64',
+        os: 'darwin',
+      },
+      name: 'darwin-amd64',
+    },
+  };
+
+  // Exact matches
+  expect(
+    imageRegistry.getBestManifest(
+      [manifests['linux-amd64'], manifests['linux-arm64'], manifests['windows-amd64'], manifests['windows-arm64']],
+      'amd64',
+      'linux',
+    ),
+  ).toHaveProperty('name', 'linux-amd64');
+  expect(
+    imageRegistry.getBestManifest(
+      [manifests['linux-amd64'], manifests['linux-arm64'], manifests['windows-amd64'], manifests['windows-arm64']],
+      'amd64',
+      'windows',
+    ),
+  ).toHaveProperty('name', 'windows-amd64');
+
+  // Linux by default
+  expect(
+    imageRegistry.getBestManifest(
+      [manifests['linux-amd64'], manifests['linux-arm64'], manifests['windows-amd64'], manifests['windows-arm64']],
+      'amd64',
+      'darwin',
+    ),
+  ).toHaveProperty('name', 'linux-amd64');
+  expect(
+    imageRegistry.getBestManifest([manifests['windows-amd64'], manifests['linux-amd64']], 'amd64', 'darwin'),
+  ).toHaveProperty('name', 'linux-amd64');
+
+  // Only one os by default
+  expect(
```

Comment:
I think here it's the blurry lines

Most of people will do Linux images so I would think it's ok to only have partial match when OCI image has Linux as platform. But for windows and Darwin I would think that it doesn't make sense to have this kind of partial matching

Probably that if you do a Darwin image then probably it means that it should not work on windows or Linux

If you do a linux/arm64 orblinux/amd64 then yes here probably you don't mean the exact os but mainly the arch

WDYT ?

Reply:
Yes, I completely agree. I'll make the change

Reply:
👍